torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -468,7 +481,161 @@ class ScriptModuleSerializer {
     }
   }
 
-  void writeByteCode(const Module& module, bool save_mobile_debug_info) {
+  //  Remove the tensor Ivalue in elements "constants" field if they exist in
```

Comment:
It's trying to parse the bytecode and there are duplicated codes in import.cpp. Why parsing is needed here? Could we get the constant table directly from the constant table, before write it to bytecode?

Reply:
Hmm can you elaborate a bit more? I'm trying to understand your suggestion. Currently `constants_from_jit` (key: constant tensor, value: index in jit constant table) is the constant table I get from jit, and it is used to parse the tensor in constant list to index ( (a) format to (b) format ) , before writing to bytecode (`writeArchive("bytecode", telements)`). 
(a) format
```
   ('constants',
    (0.1,
     1e-05,
     None,
     2,
     1,
     False,
     0,
     True,
     False,
     1,
     2,
     3,
     torch._utils._rebuild_tensor_v2(pers.obj(('storage', torch.FloatStorage, '0', 'cpu', 90944),),
       0,
       (1, 116, 28, 28),
       (90944, 784, 28, 1),
       False,
       collections.OrderedDict()))
```

to 
(b) format
```
   ('constants',
    (0.1,
     1e-05,
     None,
     2,
     1,
     False,
     0,
     True,
     False,
     1,
     2,
     3,
     ('tensor_jit_index', (4,)),
```

When import the new model in `import.cpp`, it parse the (b) format to (a) format, with the `constants_from_jit` map.

Reply:
Is the code below to find constant_values from byte code, and find if it's in the jit's constant table? constant_values can be accessed from mobile modules' code, like in https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/serialization/export_module.cpp#L213 . You don't have to get it by parsing the bytecode.

Reply:
Ah I see. Let me check how to use it..

Reply:
Update in the latest commit. I add an extra argument in `getFunctionTuple()` so it has access to `constant_from_jit`, feel free to let me know if it makes sense.